backend/cmm.mli

```diff
@@ -369,8 +369,36 @@ type alloc_dbginfo_item =
 
 type alloc_dbginfo = alloc_dbginfo_item list
 
+type is_global =
+  | Global
+  | Local
+
+val equal_is_global : is_global -> is_global -> bool
+
+(* Symbols are marked with whether they are local or global, at both definition
+   and use sites.
+
+   Symbols defined as [Local] may only be referenced within the same file, and
+   all such references must also be [Local].
+
+   Symbols defined as [Global] may be referenced from other files. References
+   from other files must be [Global], but references from the same file may be
+   [Local].
+
+   (Marking symbols in this way speeds up linking, as many references can then
+   be resolved early) *)
+type symbol =
+  { sym_name : string;
+    sym_global : is_global
+  }
+
 type operation =
-  | Capply of machtype * Lambda.region_close
+  | Capply of
+      { result_type : machtype;
+        region : Lambda.region_close;
+        callees : symbol list option
+            (* List of possible callees, or [None] if not known *)
```

Comment:
If I remeber correctly from an earlier discussion, the actual callee (whose address is the first argument of this operation) can be a re-optimized version of these functions, i.e., the name of the actual callee may not be in the `callees` list, and therefore the list cannot be used for emitting direct calls (e.g., if the list has only one element). If this is still true, it would be nice to make this caveat explicit in the name of the field or at least a comment.